### ISSUE TITLE
feat: fused softmax kernel

### ISSUE THREAD

**Issue by avuppal** [avuppal/triton-kernels#9]
Implement a fused softmax kernel using Triton with tiling. Add tests and benchmarks comparing to PyTorch.
